substrate/primitives/runtime-interface/src/pass_by.rs

```diff
@@ -424,7 +558,23 @@ impl<'a, T: codec::Encode> IntoFFIValue for PassFatPointerAndDecodeSlice<&'a [T]
 }
 
 /// A trait signifying a primitive type.
-trait Primitive: Copy {}
+trait Primitive: Copy + Default {
+	const SIZE: usize = core::mem::size_of::<Self>();
+
+	#[cfg(not(substrate_runtime))]
+	fn to_bytes(&self) -> &[u8] {
+		// SAFETY: Representing a primitive type as a byte slice of corresponding size is always
+		// safe.
+		unsafe { core::slice::from_raw_parts(self as *const _ as *const u8, Self::SIZE) }
+	}
```

Comment:
This will break when the host endianness is different to the one of the wasm runtime.

Reply:
If it works or not. You can just add a to_le_bytes call here and this would not be an issue

Reply:
While we don't really care about big endian hosts (I'm not sure whether we'd currently even compile on such host, nevermind actually running correctly; big endian is pretty much dead nowadays outside of retrocomputing) and I don't think there would ever be a case in practice where the runtime endianess is different than the host, however this conversion to slice is indeed unnecessary since you're converting that to a pointer anyway in the end, so what you'd want to do is to just return a raw pointer directly (e.g. something like `fn as_ptr(&self) -> { &raw const self }`).

Or you could do what Basti suggested and do endianness swapping, but I'm not convinced it's worth it as it's extra complexity (but if Basti would prefer it that way then I don't mind :P).

Reply:
I considered using `.to_le_bytes()` as well, I'm just not a big fan of it as it's not a trait member, but an independently implemented method, so more macro magic is needed to implement it, which is not a problem ofc, but why bother if we're not going to support BE hosts anyway?

Reply:
We can just do it right. All the time we are wasting here arguing if big endian is still used or whatever. We have all the code written in the way to handle endianess, we should continue doing so. Maybe it is useless, for sure. I'm giving you this point. But then we should just drop the support from everywhere and say it so, but if we keep this small piece here different and someone tries to run this on some BE architecture, debugging this will be hell.